### PR TITLE
Using dist.mode instead of logits.argmax. More compact.

### PR DESCRIPTION
**Pull request by arnaujc91**
changed all the occurrences where picking an action deterministically

- **from**: using the outputs of the actor network.
- **to**: using the mode of the PyTorch distribution.

This was agreed with @MischaPanch. 

Please make sure everything is alright.

### CODE REVIEW

File: tianshou/policy/modelfree/discrete_sac.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def forward(  # type: ignore
         logits, hidden = self.actor(batch.obs, state=state, info=batch.info)
         dist = Categorical(logits=logits)
         if self.deterministic_eval and not self.training:
```
Comment (MischaPanch) [inline tianshou/policy/modelfree/discrete_sac.py:111]: pls adjust the docstring of what happens when `deterministic_eval=True` is passed

**Comment by MischaPanch** [comment]
@arnaujc91 a test fails due to cryptic float/rounding errors. Could you introduce a rounding to say 5 digits to make sure the test succeeds?

File: tianshou/policy/modelfree/discrete_sac.py (hunk 1 above)
Comment (arnaujc91) [reply tianshou/policy/modelfree/discrete_sac.py:111]: done, check please

**Comment by arnaujc91** [comment]
> @arnaujc91 a test fails due to cryptic float/rounding errors. Could you introduce a rounding to say 5 digits to make sure the test succeeds?

Please check test modifications.

File: tianshou/policy/modelfree/redq.py [hunk 2]
```diff
@@ -153,7 +153,10 @@ def forward(  # type: ignore
         loc_scale, h = self.actor(batch.obs, state=state, info=batch.info)
         loc, scale = loc_scale
         dist = Independent(Normal(loc, scale), 1)
-        act = loc if self.deterministic_eval and not self.training else dist.rsample()
```
Comment (arnaujc91) [inline tianshou/policy/modelfree/redq.py:156]: I think we should extract a method out of this as it is used in multiple places. I would suggest a simple
`act = get_action()` method.

Comment (MischaPanch) [reply tianshou/policy/modelfree/redq.py:156]: Yes, you can add a method `get_action_from_dist(dist: Distribution)` to PGPolicy and SACPolicy. Pls add a docstring describing its behavior

**Comment by MischaPanch** [comment]
@arnaujc91 pls resolve the conflicts with master

**Comment by arnaujc91** [comment]
> @arnaujc91 pls resolve the conflicts with master

done

File: tianshou/policy/modelfree/redq.py (hunk 2 above)
Comment (MischaPanch) [reply tianshou/policy/modelfree/redq.py:156]: I guess you'll do it in the followup PR. Merging this
